### DISCUSSION TITLE
Subtitles (not DVD)

### DISCUSSION THREAD

**Discussion by LoisGNS** [OpenSageTV/Community#53] [Q&A]
I have some streaming shows that I used Playon to record. They have subtitles (and are otherwise in French, so I want to be able to toggle those on or off - maybe someday my French will be good enough to toggle them off!). I run my Playon recordings through Handbrake to make them more compact & eliminate unnecessary stuff at the start & end. Unless I tell HB to burn the subtitles in, I can't seem to make them show up in Sage, though VLC will display them (or not) via a menu selection. Is making HB burn the subtitles in my only option for Sage to be able to display them? When I don't need the subtitles, they are distracting, so I'd prefer to have them there but only on when I tell them to display. Am I missing a command in the extenders, placeshifter, or client that will do that? Or maybe a .properties setting? 

I tried searching the old forum, but it seems to have been made unsearchable. I also saved a local copy of the static version but searching is likewise not very effective.

**Comment by Ugadata** [comment]
The open source version of Sage isn't so good when it comes to subtitles and/or closed caption.

As you found, you can burn the subtitles in.  If you can extract the subtitles and store them as an external file (.srt), Sage can turn them on and off, somewhat on the fly.

**Reply by LoisGNS** [reply]
What tools are available to extract subtitles to separate files - and then, how to tell Sage to use them?

**Comment by Ugadata** [comment]
I am not the best source to answer this, however I can make a few suggestions.

Check if Playon will save subtitles to an external file.  I don't currently use Playon and don't remember if it can do this or not.
Not sure if Handbrake can do this.  Internet searching was more on the it can't or doesn't but maybe there is something???
MKVToolNIX and MP4Box should be able to extract subtitles.  It looks as though FFMPEG can do this as well. ( FYI -I have not used any of these for this purpose)
IIRC  Jubler is a program for editing subtitles and may have the ability to save them to an external file.

</sidenote - digression>
`I have been using AnyStream to download TV series and Movies from supported streaming services.  It is not free and does require a subscription to the streaming provider.  It does allow saving subtitles external to the video file.  Not all streaming providers are supported.   Flix, Prime, Hulu, Disney+ and a few others are supported.
Once downloaded I import them into SageTV to watch.  Obviously, this is not a Sage only option.`
</sidenote -digression:off>


Within Sage'e Detailed Setup options you can choose weather the subtitles are shown by default or not.  Either way there is also the option to turn on/off subtitles on the fly while the video is playing (like VLC).

**Reply by LoisGNS** [reply]
I found a setting for default subtitle language in Detailed Setup. Is that what you're referring to? I didn't see anything about just turning them on or off. In any event, the behavior seems inconsistent. Most of my recordings don't have any subtitles, or they're burned-in and there's no separate track to play with, but in the case of one series they were in this separate track, so I thought I'd try to get Sage to play or not play as an option. I was testing the one that started me on this "quest," and when I played it on an HD200 extender the Options button showed a screen that included a subtitles checkbox, which did  work to toggle the subtitle track (not in a separate file) on or off. But when I use a client, place-shifter, or HD100 extender, it doesn't show. 

I couldn't find any SRT export option in Handbrake, but Google did find suggestions for doing it with ffmpeg. The file in question is currently on a computer that I'm leaving off to avoid over-heating my computer room until the current heatwave abates, so I can't test anything further for a few days. I want to decide whether to burn-in the subtitles, keep them in a separate track, or try to export before I run the rest of the episodes through Handbrake. 


Thanks for the reference to AnyStream. I have looked at, but so far Playon + Audials have been good supplements to Sage for saving streaming shows. I looked at AnyStream, but it didn't seem to do much that my existing tools don't do (at least regarding the stuff I care about). Have you also tried PlayOn and/or Audials & have an opinion of what AnyStream might do better/differently/additionally?

**Comment by Ugadata** [comment]
> I found a setting for default subtitle language in Detailed Setup. Is that what you're referring to? I didn't see anything about just turning them on or off. In any event, the behavior seems inconsistent. Most of my recordings don't have any subtitles, or they're burned-in and there's no separate track to play with, but in the case of one series they were in this separate track, so I thought I'd try to get Sage to play or not play as an option. I was testing the one that started me on this "quest," and when I played it on an HD200 extender the Options button showed a screen that included a subtitles checkbox, which did work to toggle the subtitle track (not in a separate file) on or off. But when I use a client, place-shifter, or HD100 extender, it doesn't show.

Not sure?  I will need to actually look again.  What I am referencing isn't weather subtitles are on or off, so much, as how Sage behaves when it does recognize that subtitles are available.  Regardless of the option in the Detail Setup, if Sage recognizes that subtitles are present you will see a Subtitles option in the option menu to turn them on/off on the fly and an option for the Subtitle Language. 
Open source Sage TV doesn't seem to recognize subtitles , on imported video, if the subtitles are embedded in the file.  
I think back when Sage had a proprietary MPEG codec built-in it did much better job of dealing with subtitles.

**Comment by Ugadata** [comment]
I have used PlayOn (Desktop) and was aware of Audials (but never used it).
The main difference between PlayOn and Anystream is PlayOn is a screen recorder (and I believe Audials is too) and AS is a downloader.  A screenrecorder will take an hour to grab an hour show; a downloader will grab it based on how long it would take to download the file. 

I always found PlayOn to be finicky.  I typically only used PlayOn to grab shows that Sage missed recording (for whatever reason).  Sometimes PlayOn would grab the show no problem and other times it was like pulling teeth. 

AS has the option to download subtitles either embedded in the video or as a separate external file(s).

**Reply by LoisGNS** [reply]
Thanks. That's useful info about AnyStream. Audials does have a fast-mode that works for some sources. I mainly use it for shows  that Sage misses and PlayOn can't get: primarily stuff that's on Xfinity. PlayOn used to support Xfinity but no longer does so. Audials does, but not nearly as seamlessly as PO used to. If AS supported downloads from Xfinity, I'd definitely consider it, as that's what I usually need when I miss a Sage recording. Currently, PO handles everything for Netflix, Amazon, Paramount+, Peacock and Hulu, plus a few other things. It does misbehave more frequently than I'd like, though.

**Reply by Ugadata** [reply]
No, AS doesn't support Xfinity and probably never will.

**Reply by valnar1** [reply]
Install a free program called MediaInfo.  I put it on every computer.  Set the mode to "text".  Copy the output of one of your videos selectable subtitles into a comment here for us to see.  That'll tell us what sub format, and possibly why you can't view them.
